cetmix_tower_yaml/wizards/cx_tower_yaml_import_wiz.py

```diff
@@ -59,49 +50,110 @@ def action_import_yaml(self):
 
         # Parse YAML code
         yaml_data = yaml.safe_load(self.yaml_code)
+        records = yaml_data.get("records")
+        if not records:
+            raise ValidationError(_("YAML file doesn't contain any records"))
+
+        # Cache models
+        models = {}
+        odoo_record_ids = []
+
+        # Process each record
+        for record in records:
+            record_reference = record.get("reference")
+            if not record_reference:
+                raise ValidationError(_("Record reference is missing"))
+            model_name = record.get("cetmix_tower_model")
+            if not model_name:
+                raise ValidationError(
+                    _("Record model is missing for record %s", record_reference)
+                )
 
-        # Update existing record
-        if (
-            self.record_id
-            and yaml_data.get("reference")
-            and self.if_record_exists == "update"
-        ):
-            record = self.env[self.model_name].browse(self.record_id)
-            record.update({"yaml_code": self.yaml_code})
-        else:
-            model = self.env[self.model_name]
+            # Get model from cache or create new one
+            model = models.get(model_name)
+            if not model:
+                model = self.env[f"cx.tower.{model_name.replace('_', '.')}"]
+                models[model_name] = model
+
+            # Get existing record by reference
+            if self.if_record_exists == "update":
+                odoo_record = model.get_by_reference(record_reference)
+                # Update existing record
+                if odoo_record:
+                    try:
```

Comment:
```python
try:
    with cr.savepoint():
        ...
```

Reply:
since you catch an exception and do not call the caught exception in this case the transaction will not roll back automatically